mercury.spec

```diff
@@ -105,7 +83,7 @@ Mercury plugin to support the UCX transport.
         -DNA_USE_DYNAMIC_PLUGINS:BOOL=ON                  \
         -DNA_INSTALL_PLUGIN_DIR:PATH=%{_libdir}/mercury   \
         -DNA_USE_SM:BOOL=ON                               \
-        -DNA_USE_UCX:BOOL=%{ucx}                          \
+        -DNA_USE_UCX:BOOL=ON                              \
```

Comment:
We always build against ucx yes (CentOS7 was the only platform where we were not supporting ucx) but that's not what it means. We always build all plugins in a single pass now. That generates plugin libraries (see `NA_USE_DYNAMIC_PLUGINS:BOOL=ON`) for ofi and ucx which are then dynamically loaded. So there is still an additional rpm for ucx but it now only contains the ucx plugin lib (not a separate copy of the entire mercury set as it used to be).

Reply:
Is there any good reason to not have the plugin lib available in a single version of mercury?   I have no issue with this patch, by the way, just more hesitant to land prerequisite patches without visibility into internal discussions.

Reply:
Hmm I don't quite understand your question :) but I'm more than happy to walk you through things if needed. The idea behind shipping plugin libraries separately is that some systems don't come with ucx obviously so we don't want to always include ucx and we can't have multiple rpm versions of the same mercury or we'd fall back into the situation that we used to have where we'd have multiple rpms providing the same mercury target (this is what was creating all the dependency mess previously if you remember). Having said that we also always ship ofi by default and to be consistent ideally we should have a mercury-libfabric rpm that would provide the libfabric plugin library separately. No worries wrt landing, not really urgent and I understand I'll see landing questions with @mchaarawi.

Reply:
Ok, I think I misunderstood. I assumed plugin to mean a dlopen but I guess that is to the plugin not the underlying library it depends on. You could potentially remove the dependence with something like weak symbols (or dlopen) so that you could ship the plugin even if the underlying libraries are not installed.

Reply:
plugin does mean dlopen in that case. So you have all main mercury libraries libna.so libmercury.so in /usr/lib64 and then in /usr/lib64/mercury you have the plugins libna_plugin_ucx.so and libna_plugin_ofi.so. Those get dlopened by libna.so. libna_plugin_ucx.so has dependencies against ucx, libna_plugin_ofi.so has dependencies against libfabric.

Reply:
There is no reason to me for a user to install libna_plugin_ucx.so if ucx is not installed, unless I'm missing your point.